src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.AccessibilityHelper.cs

```diff
@@ -78,18 +80,26 @@ public static bool IsLessAccessibleThan(ISymbol? first, INamedTypeSymbol second)
                 }
             }
 
-            private static bool IsTypeLessAccessibleThanOtherType(ITypeSymbol? first, INamedTypeSymbol second)
+            private static bool IsTypeLessAccessibleThanOtherType(ITypeSymbol? first, INamedTypeSymbol second, List<ITypeSymbol>? alreadyCheckingTypes = null)
```

Comment:
1. Use a HashSet.
2. No need to be optional.  It means recursive calls may fail to pass this, causing a problem.